nav2_costmap_2d/include/nav2_costmap_2d/costmap_2d_ros.hpp

```diff
@@ -99,6 +99,21 @@ class Costmap2DROS : public nav2_util::LifecycleNode
     const std::string & local_namespace,
     const bool & use_sim_time);
 
+  /**
+   * @brief  Constructor for the wrapper
+   * @param name Name of the costmap ROS node
+   * @param parent_namespace Absolute namespace of the node hosting the costmap node
+   * @param local_namespace Namespace to append to the parent namespace
+   * @param use_sim_time Whether to use simulation or real time
+   * @param is_lifecycle_follower Whether launch as a child-LifecycleNode following others
+   */
+  explicit Costmap2DROS(
+    const std::string & name,
+    const std::string & parent_namespace,
+    const std::string & local_namespace,
+    const bool & use_sim_time,
+    const bool & is_lifecycle_follower);
```

Comment:
No, not a constructor argument. 3 contructors up there is `Costmap2DROS();` That is what should be modified with `NodeOptions` and set `is_lifecycle_follower_` in the method (e.g. not a argument)

Reply:
i'm not very familiar with `NodeOptions`. you mean add following code into `Costmap2DROS::init()`?
```cpp
declare_parameter("is_lifecycle_follower", rclcpp::ParameterValue(false));
```

however,, i'm not clear how to set it into `true` when declaring a costmap node
```cpp
  costmap_ros_ = std::make_shared<nav2_costmap_2d::Costmap2DROS>(
    "local_costmap", std::string{get_namespace()}, "local_costmap",
    get_parameter("use_sim_time").as_bool()); 
```
for example, in `controller_service.cpp`, how should I modify this declaration to make `is_lifecycle_follower_` being `true`?

Reply:
[I linked what you should do](https://github.com/ros-planning/navigation2/blob/4bc3e8a534005e983ee7a49a09ec00aeb9820661/nav2_planner/include/nav2_planner/planner_server.hpp#L58)

Reply:
`is_lifecycle_follower_` is a member, not a parameter. Just set it in that default contractor to `false` since that's when it is not a follower. Else, default the value in the hpp file to true since all users of other constructors are automatically not independent nodes.

Reply:
> default the value in the hpp file to true since all users of other constructors are automatically not independent nodes.

I'm not very sure about it. I default the value in the hpp file to `false` because I think it's more common to regard it as an independent node, just like how unit test does ?

Reply:
I'm so unclear about your suggested changes that I need your help to point out why our latest code is wrong....